src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -27,103 +23,78 @@ import * as ReceiptUtils from '@libs/ReceiptUtils';
 import * as ReportActionUtils from '@libs/ReportActionsUtils';
 import * as ReportUtils from '@libs/ReportUtils';
 import * as TransactionUtils from '@libs/TransactionUtils';
-import reportActionPropTypes from '@pages/home/report/reportActionPropTypes';
-import reportPropTypes from '@pages/reportPropTypes';
+import type {ContextMenuAnchor} from '@pages/home/report/ContextMenu/ReportActionContextMenu';
 import * as IOU from '@userActions/IOU';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {Policy, Report, ReportAction, Session} from '@src/types/onyx';
+import type DeepValueOf from '@src/types/utils/DeepValueOf';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import ReportActionItemImages from './ReportActionItemImages';
 
-const propTypes = {
-    /** All the data of the action */
-    action: PropTypes.shape(reportActionPropTypes).isRequired,
-
-    /** The associated chatReport */
-    chatReportID: PropTypes.string.isRequired,
-
-    /** The active IOUReport, used for Onyx subscription */
-    // eslint-disable-next-line react/no-unused-prop-types
-    iouReportID: PropTypes.string.isRequired,
+type PaymentVerbTranslationPath = 'iou.payerSpent' | 'iou.payerOwes' | 'iou.payerPaid';
 
-    /** The report's policyID, used for Onyx subscription */
-    policyID: PropTypes.string.isRequired,
+type PaymentMethodType = DeepValueOf<typeof CONST.IOU.PAYMENT_TYPE>;
```

Comment:
maybe also we can reuse this type in other places like `ReportUtils` and `OriginalMessage`